core/templates/components/entity-creation-services/collection-creation.service.spec.ts

```diff
@@ -0,0 +1,151 @@
+    loaderService = TestBed.get(LoaderService);
+    analyticsService = TestBed.get(SiteAnalyticsService);
+    windowRef = TestBed.get(WindowRef);
+    httpTestingController = TestBed.get(HttpTestingController);
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  it('should successfully create a collection and navigate to collection',
+    fakeAsync(() => {
+      spyOn(alertsService, 'clearWarnings').and.callThrough();
+      spyOn(loaderService, 'showLoadingScreen').and.callThrough();
+      spyOn(analyticsService, 'registerCreateNewCollectionEvent')
+        .and.callThrough();
+
+      spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+        location: {
+          href: ''
+        }
+      });
+
+      collectionCreationService.createNewCollection();
+
+      let req = httpTestingController.expectOne(
+        '/collection_editor_handler/create_new');
+      expect(req.request.method).toEqual('POST');
+      req.flush({collectionId: SAMPLE_COLLECTION_ID});
+
+      flushMicrotasks();
+      tick(150);
+
+      expect(alertsService.clearWarnings).toHaveBeenCalled();
+      expect(loaderService.showLoadingScreen)
+        .toHaveBeenCalledWith('Creating collection');
+      expect(analyticsService.registerCreateNewCollectionEvent)
+        .toHaveBeenCalledWith(SAMPLE_COLLECTION_ID);
+
+      expect(windowRef.nativeWindow.location.href).toEqual(
+        '/collection_editor/create/' + SAMPLE_COLLECTION_ID);
+    })
+  );
+
+  it('should fail to create a collection and hide the loading screen',
+    fakeAsync(() => {
+      spyOn(loaderService, 'hideLoadingScreen').and.callThrough();
+
+      collectionCreationService.createNewCollection();
+
+      let req = httpTestingController.expectOne(
+        '/collection_editor_handler/create_new');
+      expect(req.request.method).toEqual('POST');
+      req.flush('Error creating a new collection.', {
+        status: ERROR_STATUS_CODE,
+        statusText: 'Error creating a new collection.'
+      });
+
+      flushMicrotasks();
+
+      expect(loaderService.hideLoadingScreen).toHaveBeenCalled();
+    })
+  );
+
+  it('should not be able to be used while in progress',
+    fakeAsync(() => {
+      spyOn(alertsService, 'clearWarnings').and.callThrough();
+      spyOn(loaderService, 'showLoadingScreen').and.callThrough();
+      spyOn(analyticsService, 'registerCreateNewCollectionEvent')
+        .and.callThrough();
+
+      spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+        location: {
+          href: ''
+        }
+      });
+
+      collectionCreationService.createNewCollection();
+      collectionCreationService.createNewCollection();
+
+      let req = httpTestingController.expectOne(
+        '/collection_editor_handler/create_new');
+      expect(req.request.method).toEqual('POST');
+      req.flush({collectionId: SAMPLE_COLLECTION_ID});
+
+      flushMicrotasks();
+      tick(150);
+
+      expect(alertsService.clearWarnings)

```

Comment:
I don't think you should test if AlertsService is called

Reply:
Fixed.